--- imgui_impl_metal.mm.orig	2026-09-12 02:54:13
+++ imgui_impl_metal.mm	2026-09-25 13:59:42
@@ -146,7 +146,7 @@
 {
     ImGui_ImplMetal_Data* bd = ImGui_ImplMetal_GetBackendData();
     IM_ASSERT(bd != nil && "Context or backend not initialized! Did you call ImGui_ImplMetal_Init()?");
-#ifdef IMGUI_IMPL_METAL_CPP
+#if !__has_feature(objc_arc)
     bd->SharedMetalContext.framebufferDescriptor = [[[FramebufferDescriptor alloc] initWithRenderPassDescriptor:renderPassDescriptor]autorelease];
 #else
     bd->SharedMetalContext.framebufferDescriptor = [[FramebufferDescriptor alloc] initWithRenderPassDescriptor:renderPassDescriptor];
@@ -407,7 +407,7 @@
     bd->SharedMetalContext.samplerStateNearest = [device newSamplerStateWithDescriptor:samplerDescriptor];
 
     ImGui_ImplMetal_CreateDeviceObjectsForPlatformWindows();
-#ifdef IMGUI_IMPL_METAL_CPP
+#if !__has_feature(objc_arc)
     [samplerDescriptor release];
     [depthStencilDescriptor release];
 #endif
